test/src/tests-identity.ts

```diff
@@ -4355,6 +4362,145 @@ describe('identity', function() {
+            } as IKitConfigs;
+
+            roktKit = {
+                filters: {
+                    userAttributeFilters: [],
+                    filterUserAttributes: () => ({}),
+                    filteredUser: null,
+                },
+                launcher: () => {},
+            } as unknown as IRoktKit;
+        });
+
+        it('should set currentUser once the email is positively identified', async () => {
+            fetchMockSuccess(urls.identify, {
+                mpid: testRoktMPID,
+                is_logged_in: false,
+                matched_identities: {
+                    email: testRoktEmail,
+                },
+            });
+
+            mParticle.init(apiKey, { ...window.mParticle.config, kitConfigs: [roktConfig] });
+
+            await waitForCondition(() => hasIdentifyReturned(testRoktMPID));
+
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._RoktManager.attachKit(roktKit);
+            
+            mParticle.Identity.identify({
+                userIdentities: {
+                    email: testRoktEmail,
+                },
+            });
+
+            await waitForCondition(() => hasIdentifyReturnedWithEmail(testRoktEmail));
+
+            expect(mpInstance._RoktManager['currentUser']).to.not.be.null;
+            expect(mpInstance._RoktManager['currentUser'].getUserIdentities().userIdentities.email).to.equal(testRoktEmail);
+        });
+
+        it('should not set currentUser if the initial identify call is not successful', async () => {
+            fetchMock.post(urls.identify, {
+                status: HTTP_BAD_REQUEST,
+                body: {
+                    Errors: [
+                        {
+                            message: 'knownIdentities is empty.',
+                            code: 'LOOKUP_ERROR'
+                        },
+                    ],
+                    ErrorCode: 'LOOKUP_ERROR',
+                    StatusCode: HTTP_BAD_REQUEST,
+                    RequestId: '123',
+                },
+            }, {
+                overwriteRoutes: true,
+            });
+
+            mParticle.init(apiKey, { ...window.mParticle.config, kitConfigs: [roktConfig] });
+
+            await waitForCondition(hasIdentityCallInflightReturned);
+
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._RoktManager.attachKit(roktKit);
+
+            await waitForCondition(hasIdentityCallInflightReturned);
+
+            expect(mpInstance._RoktManager['currentUser']).be.null;
+        })
+
+        it('should not set currentUser if the email is not positively identified', async () => {
+            fetchMockSuccess(urls.logout, {
+                context: null,
+                matched_identities: {
+                    device_application_stamp: 'my-das',
+                },
+                is_ephemeral: true,
+                mpid: testMPID,
+                is_logged_in: false,
+            });
+
```

Comment:
is there a reason to mock `urls.logout` here if you are not calling `logout` below?

Reply:
Nope. That's a bad copypasta